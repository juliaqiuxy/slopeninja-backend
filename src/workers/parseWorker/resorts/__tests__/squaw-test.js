import fs from 'fs';
import {
  parseSquawSnow,
  parseSquawLiftCounts,
  parseSquawLifts,
  parseSquawTrails,
  parseSquawTrailCounts,
} from '../squaw';
import { createHtmlParser } from '../../parserFactory';

test('fetches Squaw snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/squaw-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseSquawSnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: 'sunny',
      temperature: 32,
      baseCondition: null,
      newSnow: 7,
      snowDepthBase: 18,
      snowDepthSummit: 23,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseSquawSnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Squaw lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('liftCounts', parseSquawLiftCounts)(htmlText);
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: 1,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('liftCounts', parseSquawLiftCounts)('<html></html>');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Squaw lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('lifts', parseSquawLifts)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('lifts', parseSquawLifts)('<html></html>');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Squaw trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('trails', parseSquawTrails)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('trails', parseSquawTrails)('<html></html>');
  expect(resortData).toMatchObject({ trails: [] });
});


test('fetches Squaw trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/squaw-lifts.html.input`);
  const resortData = await createHtmlParser('trailCounts', parseSquawTrailCounts)(htmlText);
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: 0,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trailCounts', parseSquawTrailCounts)('<html></html>');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});
