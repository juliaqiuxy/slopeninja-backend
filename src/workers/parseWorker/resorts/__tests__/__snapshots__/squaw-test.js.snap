// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Squaw lift list correctly 1`] = `
Object {
  "lifts": Array [],
}
`;

exports[`fetches Squaw trail list correctly 1`] = `
Object {
  "trails": Array [],
}
`;
